src/main/scala/com/microsoft/partnercatalyst/fortis/spark/StreamsChangeListener.scala

```diff
@@ -0,0 +1,66 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+import java.time.Duration
+import java.util.concurrent.CompletableFuture
+import java.util.{Timer, TimerTask}
+
+import com.microsoft.azure.servicebus._
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
+import org.apache.spark.streaming.StreamingContext
+
+object StreamsChangeListener {
+
+  @volatile @transient private var queueClient: QueueClient = _
```

Comment:
This object won't be serialized so you should be able to make this a `val` right? In that way, it should also be safe to access from the inner class helper thread.